### PR TITLE
Update for changes to HTML's structured cloning/transferring

### LINKED ISSUE #701: Update for HTML structured clone changes
https://github.com/whatwg/html/commit/97d644c97335956610a31e8ad98d1a388c063e84 changed structured cloning significantly. We need to:

- Not use StructuredClone(), but instead StructuredDeserialize(StructuredSerialize())
- Roll our own Transfer() for array buffers, as Transfer() is no longer defined in HTML.

### PR DESCRIPTION
**Pull request by domenic**
Closes #701.

---

This is basically editorial so if people are busy no need to review this; I'll merge it in a few days. But as always review is nice.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -797,11 +795,14 @@ readable stream is <a>locked to a reader</a>.
...
 stream">tee</a> a given readable stream.
 
-The second argument, <var>cloneForBranch2</var>, governs whether or not the data from the original stream will be <a
-abstract-op lt="StructuredClone">structured cloned</a> before appearing in the second of the returned branches. This is
+The second argument, <var>cloneForBranch2</var>, governs whether or not the data from the original stream will be cloned
+(using HTML's <a>serializable objects</a> framework) before appearing in the second of the returned branches. This is
 useful for scenarios where both branches are to be consumed in such a way that they might otherwise interfere with each
-other, such as by <a abstract-op lt="Transfer">transfering</a> their <a>chunks</a>. However, it does introduce a
-noticable asymmetry between the two branches. [[!HTML]]
+other, such as by <a href="#transfer-array-buffer">transfering</a> their <a>chunks</a>. However, it does introduce a
```
Comment (isonmad) [inline index.bs]: This defines transfering by linking to TransferArrayBuffer, which asserts its argument has an [[ArrayBufferData]] internal slot, but this is about cloning arbitrary objects.

Comment (ricea) [inline index.bs]: Should be "transferring"?

File: reference-implementation/lib/helpers.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ exports.PromiseInvokeOrPerformFallback = (O, P, args, F, argsF) => {
 };
 
 // Not implemented correctly
-exports.SameRealmTransfer = O => O;
+exports.TransferArrayBuffer = O => O;
```
Comment (ricea) [inline reference-implementation/lib/helpers.js]: How about using O => O.slice(0) instead? Or has this already been discussed?

**Comment by ricea** [comment]
lgtm

File: index.bs (hunk 1 above)
Comment (domenic) [inline index.bs]: This sub-clause of the sentence is "such as", so yeah, it's just giving an example of how branches could interfere with each other.

File: reference-implementation/lib/helpers.js (hunk 2 above)
Comment (domenic) [inline reference-implementation/lib/helpers.js]: In general this could probably be a lot better; that's a reasonable start.

File: index.bs (hunk 1 above)
Comment (domenic) [inline index.bs]: I'll link to HTML's "transferable objects" instead

**Pull request merged**